[PyTorch][Static Runtime] Support memory planning for to without requiring copying

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #69854
* #69853
* #69852
* #68122
* __->__ #67223
* #67221
* #67394

Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

swolchok committed:
[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying



Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

eellison reviewed (commented):
@navahgar is working on a first invocation of SR that would give dtype/device on the tensors after that we should be able to statically tell whether `to` is a no-op or will copy. would it be easier to integrate with those changes than add all this ?

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch][Static Runtime] WIP: support memory planning for to without requiring copying"


Differential Revision: [D31776259](https://our.internmc.facebook.com/intern/diff/D31776259/)

[ghstack-poisoned]
